inference-engine/src/vpu/graph_transformer/src/middleend/passes/eliminate_redundant_conversions.cpp

```diff
@@ -29,6 +29,17 @@ void PassImpl::runForStage(const Model& model, const Stage& convert) {
     const auto input = convert->input(0);
     const auto output = convert->output(0);
 
+    //
+    // Check and remove the convert that was added to unused input
+    // In this case we will have the converted intermediate data object which is not consumed
+    //
+
+    if (output->usage() == DataUsage::Intermediate && !output->isConsumed()) {
```

Comment:
Should we check that producer is `Convert` stage?

Reply:
I don't think so, it's `runForStage` pass which is targeted to particular `StageType`, so there can be only `Convert` stage.

Reply:
Thanks for explanation